actors/builtin/miner/miner_actor.go

```diff
@@ -1126,8 +1140,11 @@ func (a Actor) DeclareFaultsRecovered(rt Runtime, params *DeclareFaultsRecovered
 
 	store := adt.AsStore(rt)
 	var st State
+	debtToBurn := abi.NewTokenAmount(0)
 	rt.State().Transaction(&st, func() {
-		verifyPledgeMeetsInitialRequirements(rt, &st)
+		// Verify unlocked funds cover both InitialPledgeRequirement and FeeDebt
+		// and repay fee debt now.
+		debtToBurn = verifyPledgeRequirementsAndRepayDebts(rt, &st)
```

Comment:
I'm a little surprised we don't vest before this. Is it because we know vesting happened in prove commit within the same epoch?

Reply:
> Is it because we know vesting happened in prove commit within the same epoch?

Did you think this was `ConfirmSectorProofsValid`?  It's `DeclareFaultsRecovered` so there is no relationship with a prove commit.  

We could vest before but in the past we haven't vested during these verifications.  See the comment above `verifyPledgeRequirementsAndPayDebts` that I lifted and modified from @anorth's original comment in the ProveCommit invocation of verification.  

From that comment: "Computing vesting here would be almost always redundant since vesting is quantized to ~daily units".  It doesn't seem like we are consistent about that everywhere though and maybe vesting has become less expensive?  @anorth does the infrequent vesting still make sense?

Reply:
Yeah, I was confused, and that's why I thought ConfirmSectorProofsValid needed to be tested. It does seem like we're vesting inconsistently.

Reply:
Infrequent still makes sense. We now do it every deadline, so any additional calls are likely redundant. It's worth still doing on withdrawal, though.